HLTrigger/JetMET/interface/HLTHFJetCleaner.h

```diff
@@ -0,0 +1,224 @@
+    if ((jet.pt() < jetPtThresh_) | (!withinEtaRange)) {
+      cleaned_jets->push_back(jet);
+      continue;
+    }
+
+    JetRef const jetRef(jets, ijet);
+
+    // Extract HF shape variables from the ValueMap inputs
+    float sigmaEtaEta = (*sigmaEtaEtas)[jetRef];
+    float sigmaPhiPhi = (*sigmaPhiPhis)[jetRef];
+    int centralEtaStripSize = (*centralEtaStripSizes)[jetRef];
+
+    // Cuts related to eta and phi width of HF showers:
+    // Sigma eta-phi based cut + central strip size cut
+    // We'll apply the sigma eta+phi dependent cut only if requested via the config file
+    if (applySigmaEtaPhiCut_) {
+      // Sigma eta eta and sigma phi phi
+      bool passSigmaEtaPhiCut = sigmaEtaEta - sigmaPhiPhi < sigmaEtaPhiDiffThresh_;
+
+      if (applyCornerCut_) {
+        bool inCorner = (sigmaEtaEta < cornerCutSigmaEtaEta_) & (sigmaPhiPhi < cornerCutSigmaPhiPhi_);
+        passSigmaEtaPhiCut = passSigmaEtaPhiCut & (!inCorner);
+      }
+      if (!passSigmaEtaPhiCut) {
+        continue;
+      }
+    }
+
+    if (applyStripSizeCut_) {
+      // Cut related to central strip size
+      bool passCentralStripSizeCut = centralEtaStripSize < centralEtaStripSizeThresh_;
+      if (!passCentralStripSizeCut) {
+        continue;
+      }
+    }
+
+    cleaned_jets->push_back(jet);
+  }
+
+  iEvent.put(std::move(cleaned_jets));
+}
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+template <typename JetType>
+void HLTHFJetCleaner<JetType>::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  //The following says we do not know what parameters are allowed so do no validation
+  // Please change this to state exactly what you do use, even if it is no parameters
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag, std::string>("jets", edm::InputTag("hltAK4PFJetsTightIDCorrected"));
+  desc.add<edm::InputTag, std::string>("mets", edm::InputTag("hltMet"));
+  desc.add<edm::InputTag, std::string>("sigmaEtaEta", edm::InputTag("hltHFJetShowerShape", "sigmaEtaEta"));
+  desc.add<edm::InputTag, std::string>("sigmaPhiPhi", edm::InputTag("hltHFJetShowerShape", "sigmaPhiPhi"));
+  desc.add<edm::InputTag, std::string>("centralEtaStripSize",
+                                       edm::InputTag("hltHFJetShowerShape", "centralEtaStripSize"));
+  desc.add<double, std::string>("jetPtThresh", 100);
+  desc.add<double, std::string>("dphiJetMetThresh", 2.5);
+  desc.add<double, std::string>("jetEtaMin", 2.9);
+  desc.add<double, std::string>("jetEtaMax", 5.0);
+  desc.add<double, std::string>("sigmaEtaPhiDiffThresh", 0.05);
+  desc.add<double, std::string>("cornerCutSigmaEtaEta", 0.02);
+  desc.add<double, std::string>("cornerCutSigmaPhiPhi", 0.02);
+  desc.add<int, std::string>("centralEtaStripSizeThresh", 3);
+  desc.add<bool, std::string>("applyCornerCut", true);
+  desc.add<bool, std::string>("applySigmaEtaPhiCut", true);
+  desc.add<bool, std::string>("applyStripSizeCut", true);
```

Comment:
I actually don't know what the 2nd template arg does (here `std::string`). I would say it can be removed.

Ideally, you could add comments to explain the purpose of each parameter: [`example`](https://github.com/cms-sw/cmssw/blob/3ca0032ab77b75be849b9a3a603a52e6d16ebdf0/HLTrigger/JetMET/plugins/HLTScoutingPrimaryVertexProducer.cc#L69).

Reply:
Thanks! I saw this from the doxygen page [here](https://cmsdoxygen.web.cern.ch/cmsdoxygen/CMSSW_12_4_0_pre2/doc/html/d2/dcb/classedm_1_1ParameterSetDescription.html), but I believe it works without the second template argument. I removed that and commented the parameters.